eduNEXT/eox-nelp
# Jlc/course experience api

## Description

Add course experience API based in ADR:
- LikeDislike course endpoint.
- Report course endpoint.
- LikeDislike unit endpoint.
- Report unit endpoint.


Useful information to include:
- using django restframworkjson API: 
djangorestframework-jsonapi==5.0.0 to be compatible with django rest framework of maple edx-platform.
- added tests for the API for the 4 endpoints.
- Improved coverage to 88.

## Testing instructions
Clone eox-nelp or reinstall and check the new endpoints.
Check the endpoints form in lms domain with route: `/eox-nelp/api/experience/v1/`

### After
![Peek 2023-03-08 15-32](https://user-images.githubusercontent.com/51926076/223844795-f9d4e866-de40-414d-a6bd-3645442ce40a.gif)

## Additional information

**Jirastories**
- https://edunext.atlassian.net/browse/NELP-342
- https://edunext.atlassian.net/browse/NELP-340
- https://edunext.atlassian.net/browse/NELP-346

## Checklist for Merge

- [0.5] Tested in a remote environment the like-dislike. making new image with course and report endpoints.
